Firmware/Marlin_main.cpp

```diff
@@ -1502,7 +1502,9 @@ void setup()
 #ifdef SNMM
 	if (eeprom_read_dword((uint32_t*)EEPROM_BOWDEN_LENGTH) == 0x0ffffffff) { //bowden length used for SNMM
 	  int _z = BOWDEN_LENGTH;
-	  for(int i = 0; i<4; i++) EEPROM_save_B(EEPROM_BOWDEN_LENGTH + i * 2, &_z);
+	  for(uint8_t i = 0; i < 4; i++) {
+	  	eeprom_update_word((uint16_t*)EEPROM_BOWDEN_LENGTH + i * 2, _z);
```

Comment:
Isn't this a different address than before? 
I'm not sure about the casting and pointer arithmetics rules in this case, but I'm afraid the result will produce addresses:
say we have EEPROM_BOWDEN_LENGTH = 0 for simplicity;
`(uint16_t*)EEPROM_BOWDEN_LENGTH + i * 2 -> (0 + sizeof(uint16_t) * i * 2)`
and not the original:
`(0 + i * 2)`

Reply:
That’s an interesting point. I actually am not sure so I would need to research to confirm. My instinct says this isn’t an issue, but I want to be sure. 

Am I understanding you correctly that you refer yo the casting from `uint8_t` to `uint16_t*`?

It is desireable to use `uint16_t*` to use the eeprom function correctly. https://www.nongnu.org/avr-libc/user-manual/group__avr__eeprom.html


`void eeprom_update_word (uint16_t *__p, uint16_t __value)`

Reply:
@DRracer fixed :)